src/librustc_mir/monomorphize/collector.rs

```diff
@@ -586,6 +586,14 @@ impl<'a, 'tcx> MirVisitor<'tcx> for MirNeighborCollector<'a, 'tcx> {
                     self.output.push(create_fn_mono_item(instance));
                 }
             }
+            mir::Rvalue::ThreadLocalRef(def_id) => {
```

Comment:
Huh, why didn't we need this before this PR?

Reply:
We did, it just wasn't getting caught in tests. I added src/test/ui/tls.rs which fails on current master.

Reply:
Oh darn. Thanks! Can you pull that change out into a separate PR so we can backport it to beta?

Reply:
#73065

Reply:
Thanks!